R/getWindows.R

```diff
@@ -0,0 +1,99 @@
+#' Computes sliding LD-based windows.
+#'
+#' For each SNP present in a genotpe matrix, obtains nearby SNPs in LD based on genotype correlations.
+#'
+#' @param X A matrix-like object, typically `@@geno` of a [BGData-class]
+#' object.
+#' @param range Columns of `X` to compute windows. By default, does not attempt to obtain LD windows surrounding
+#' each element of `X`.
+#' @param rSq R-squared threshold for determining genotype correlations.
+#' @param maxGaps The number of consequtive SNPs that may fall below `rSq` until LD is considered lost
+#' @return A list the length of the number of columns of `X` containing sliding LD-based windows. The elements of
+#' the list are named for each "center" SNP used in LD determination. Each element is a vector of SNP names--SNPs
+#' in LD with the "center" SNP.
+#' @export
+getWindows <- function(X, range = 1:100, rSq = 0.1, maxGaps = 2) {
+    Xwin <- X[, range]
+    win <- lapply(1:ncol(Xwin), function(i) getBlock(Xwin, i, rSq = r2, maxGaps = gaps))
```

Comment:
`r2` should be `rSq`.

Reply:
Oops yep thanks for the `r2` and `gaps` mistakes!